tests/Unit/Evolution/Systems/Cce/Actions/Test_ScriObserveInterpolated.cpp

```diff
@@ -171,6 +184,119 @@ struct test_metavariables {
+                            quadratic_coefficient * square(time));
+  const auto expected_psi_3 =
+      get(get<Tags::ScriPlus<Tags::Psi3>>(random_values)).data() * time_factor;
+  const auto expected_psi_4 =
+      get(get<Tags::TimeIntegral<Tags::ScriPlus<Tags::Psi4>>>(random_values))
+          .data() *
+      (linear_coefficient + 2.0 * quadratic_coefficient * time);
+  const auto eth_u =
+      get(get<Tags::EthInertialRetardedTime>(random_values)).data() *
+      time_factor;
+  return get(get<Tags::ScriPlus<Tags::Psi2>>(random_values)).data() *
+             time_factor +
+         expected_psi_3 * eth_u + 0.25 * expected_psi_4 * square(eth_u);
+}
+
+ComplexDataVector compute_expected_field(
+    const Variables<typename test_metavariables::cce_scri_tags>& random_values,
+    const double linear_coefficient, const double quadratic_coefficient,
+    const double time, Tags::ScriPlus<Tags::Psi1> /*meta*/) noexcept {
+  const auto time_factor = (1.0 + linear_coefficient * (time) +
+                            quadratic_coefficient * square(time));
+  const auto expected_psi_2 =
+      get(get<Tags::ScriPlus<Tags::Psi2>>(random_values)).data() * time_factor;
+  const auto expected_psi_3 =
+      get(get<Tags::ScriPlus<Tags::Psi3>>(random_values)).data() * time_factor;
+  const auto expected_psi_4 =
+      get(get<Tags::TimeIntegral<Tags::ScriPlus<Tags::Psi4>>>(random_values))
+          .data() *
+      (linear_coefficient + 2.0 * quadratic_coefficient * time);
+  const auto eth_u =
+      get(get<Tags::EthInertialRetardedTime>(random_values)).data() *
+      time_factor;
+  return get(get<Tags::ScriPlus<Tags::Psi1>>(random_values)).data() *
+             time_factor +
+         1.5 * expected_psi_2 * eth_u + 0.75 * expected_psi_3 * square(eth_u) +
+         0.125 * expected_psi_4 * pow<3>(eth_u);
+}
+
+ComplexDataVector compute_expected_field(
+    const Variables<typename test_metavariables::cce_scri_tags>& random_values,
+    const double linear_coefficient, const double quadratic_coefficient,
+    const double time, Tags::ScriPlus<Tags::Psi0> /*meta*/) noexcept {
+  const auto time_factor = (1.0 + linear_coefficient * (time) +
+                            quadratic_coefficient * square(time));
+  const auto expected_psi_1 =
+      get(get<Tags::ScriPlus<Tags::Psi1>>(random_values)).data() * time_factor;
+  const auto expected_psi_2 =
+      get(get<Tags::ScriPlus<Tags::Psi2>>(random_values)).data() * time_factor;
+  const auto expected_psi_3 =
+      get(get<Tags::ScriPlus<Tags::Psi3>>(random_values)).data() * time_factor;
+  const auto expected_psi_4 =
+      get(get<Tags::TimeIntegral<Tags::ScriPlus<Tags::Psi4>>>(random_values))
+          .data() *
+      (linear_coefficient + 2.0 * quadratic_coefficient * time);
+  const auto eth_u =
+      get(get<Tags::EthInertialRetardedTime>(random_values)).data() *
+      time_factor;
+  return get(get<Tags::ScriPlus<Tags::Psi0>>(random_values)).data() *
```

Comment:
could all of these NP quantities be tested with python instead of reimplementing the c++?